src/shared/popups/gokb-edit-platform-popup/gokb-edit-platform-popup.vue

```diff
@@ -0,0 +1,230 @@
+        selectedItem: undefined,
+        successMsg: undefined,
+        errorMsg: undefined,
+        updateUrl: undefined,
+        platform: {
+          label: undefined,
+          url: undefined
+        },
+        items: []
+      }
+    },
+    computed: {
+      localValue: {
+        get () {
+          return this.value || true
+        },
+        set (localValue) {
+          this.$emit('input', localValue)
+        }
+      },
+      isReadonly () {
+        return !accountModel.loggedIn() || !accountModel.hasRole('ROLE_EDITOR') || (this.isEdit && !this.updateUrl)
+      },
+      isEdit () {
+        return !!this.id
+      },
+      isValid () {
+        return !!this.platformItem.component
+      },
+      localSuccessMessage () {
+        return this.successMsg ? this.$i18n.t(this.successMsg, [this.$i18n.tc('component.platform.label')]) : undefined
+      },
+      localErrorMessage () {
+        return this.errorMsg ? this.$i18n.t(this.errorMsg, [this.$i18n.tc('component.platform.label')]) : undefined
+      }
+    },
+    async created () {
+      this.selectedItem = this.selected
+
+      if (this.selected) {
+        this.id = this.selected.id
+        this.fetch(this.id)
+      }
+
+      if (this.component) {
+        this.platformItem.component = this.component
+      }
+    },
+    methods: {
+      close () {
+        this.localValue = false
+      },
+      async fetch (pid) {
+        const {
+          data: {
+            label,
+            url,
+            _links
+          }
+        } = await this.catchError({
+          promise: platformServices.getPlatform(pid, this.cancelToken.token),
+          instance: this
+        })
+        this.platformItem.label = label
+        this.platformItem.url = url
+        this.updateUrl = _links?.update?.href || undefined
+      },
+      async save () {
+        const activeGroup = accountModel.activeGroup()
+
+        const newPlatform = {
+          id: this.id,
+          status: this.platformItem.status?.id || null,
+          version: this.version,
+          activeGroup
+        }
+
+        const response = await this.catchError({
+          promise: platformServices.createOrUpdatePlatform(newPlatform, this.cancelToken.token),
+          instance: this
+        })
+
+        if (response?.status === 200) {
+          this.$emit('edit', this.selectedItem)
+          this.close()
+        } else if (response?.status === 201) {
+          this.$emit('added', response.data)
+          this.close()
+        } else {
+          if (response.status === 409) {
+            this.errorMsg = 'error.update.409'
+          } else if (response.status === 500) {
+            this.errorMsg = 'error.general.500'
+          } else {
+            this.errors = response.data.error
+          }
+        }
+      }
+    }
+  }
+</script>
+
+<style>
+  .vjs-tree__node.is-highlight, .vjs-tree__node:hover {background-color:#9b9b9b }
+  .vjs-value__string { color: #3c804b }
+</style>
```

Comment:
This style block is only necessary for reviews, not platforms.

Reply:
Removed.